### PR TITLE
Implements reloader tests using KUTTL 

### PR DESCRIPTION
**Pull request by faiq**
Writes a test for reloader that checks to see if it was able to install correctly. Then creates a deployment of 50 pods that have 10 config maps each, does changes on those configmaps, and checks to ensure reloader doesn't crash. The reason we want this test is because we  want to test our memory and cpu constraints

### CODE REVIEW

File: run-tests.sh [hunk 1]
```diff
@@ -48,18 +48,18 @@ fi
 ADDON_LIST=()
 while IFS=  read -r -d $'\0'; do
     ADDON_LIST+=("$REPLY")
-done < <(find "${KUBEADDONS_REPO}/addons" -mindepth 2 -maxdepth 2 -type d -print0)
+done < <(find "${KUBEADDONS_REPO}/addons" -maxdepth 2 -type d -print0)
 
 KUBEADDONS_ABS_PATH=$(realpath -L "${KUBEADDONS_REPO}")
 export KUBEADDONS_ABS_PATH
+echo ${ADDON_LIST[*]}
```
Comment (joejulian) [inline run-tests.sh]: Was this for debugging?

File: run-tests.sh [hunk 2]
```diff
@@ -48,18 +48,18 @@ fi
 ADDON_LIST=()
 while IFS=  read -r -d $'\0'; do
     ADDON_LIST+=("$REPLY")
-done < <(find "${KUBEADDONS_REPO}/addons" -mindepth 2 -maxdepth 2 -type d -print0)
+done < <(find "${KUBEADDONS_REPO}/addons" -maxdepth 2 -type d -print0)
```
Comment (joejulian) [inline run-tests.sh]: What's this for? I thought we were requiring that depth for a reason.

File: Makefile [hunk 3]
```diff
@@ -35,13 +35,11 @@ bin/kubectl-kuttl_$(KUTTL_VERSION):
...
 	ln -sf ./kubectl-kuttl_$(KUTTL_VERSION) bin/kubectl-kuttl
 
-.PHONY: install-bin
-install-bin: bin/kind bin/kubectl-kuttl
 
 .PHONY: create-kind-cluster
 create-kind-cluster: $(KUBEADDONS_TEST_KUBECONFIG)
 
-$(KUBEADDONS_TEST_KUBECONFIG): install-bin
+$(KUBEADDONS_TEST_KUBECONFIG): bin/kind bin/kubectl-kuttl
```
Comment (joejulian) [inline Makefile]: Let's do these Makefile changes in a different PR so it's clearer that this is specifically a new test.

File: tests/kubernetes-base-addons/kuttl-test.yaml [hunk 4]
```diff
@@ -1,21 +1,7 @@
----
```
Comment (joejulian) [inline tests/kubernetes-base-addons/kuttl-test.yaml]: Unless these are required for this test, let's do this in a different PR so there can be discussion around the merits of these changes.

File: run-tests.sh [hunk 5]
```diff
@@ -48,18 +48,18 @@ fi
...
+echo ${ADDON_LIST[*]}
 
 for i in ${ADDON_LIST[*]}
 do
     printf "addon directory detected: %s\n" "${i}"
     if [ ! -d "${KUBEADDONS_TESTS_PATH}/${i}" ]
     then
-      printf "missing tests for addon directory %s\n" "${KUBEADDONS_TESTS_PATH}/${i}"
-      exit 1
+      printf "missing tests for addon directory %s\n" "${KUBEADDONS_TESTS_PATH}/${i} -- ignoring"
```
Comment (joejulian) [inline run-tests.sh]: These were not being ignored on purpose.

File: run-tests.sh (hunk 2 above)
Comment (faiq) [reply run-tests.sh]: addresesd via #13 the tests didn't run before

File: Makefile (hunk 3 above)
Comment (faiq) [reply Makefile]: touched on via #13

File: tests/kubernetes-base-addons/kuttl-test.yaml (hunk 4 above)
Comment (faiq) [reply tests/kubernetes-base-addons/kuttl-test.yaml]: removed

**Review by joejulian — APPROVED** [review]

**Review by orsenthil — APPROVED** [review]
Were these tests be generated? ( like run a command, and copy the output for the kuttl to exercise?) I wonder if you had to write this yaml?

**Comment by dlipovetsky** [comment]
This is a good example of a precisely scoped Addons test.

My only request would be to consider making the ConfigMap size more realistic, or perhaps even unrealistically large, so that we can use this test not only to tell us that reloader works with the limits, but to give us an idea of the upper limit of the resources it can reconcile. I'm currently working on https://jira.d2iq.com/browse/D2IQ-73464, an issue that demonstrates that resource size and number has a meaningful impact on controllers.
